addons/sale/models/product_template.py

```diff
@@ -290,9 +290,13 @@ def _is_add_to_cart_possible(self, parent_combination=None):
         """
         self.ensure_one()
         if not self.active:
+            # for performance: avoid calling `_get_possible_combinations`
+            return False
+        try:
+            next(self._get_possible_combinations(parent_combination))
```

Comment:
next(iterator[, default])
> Retrieve the next item from the iterator by calling its __next__() method. If **default** is given, it is returned if the iterator is exhausted, otherwise StopIteration is raised.

Reply:
I don't want to return the return of the iteration here. The result might be an empty combination that is falsy.  I need to return True specifically.

Reply:
I dont suggest to return next(xxx,)

but just replace the try except by 

if next(xxx, default):
   return what you want

else continue like before

Reply:
Following our discussion, using `next(gen, False) is not False` instead